src/components/Pagination/Pagination.tsx

```diff
@@ -1,45 +1,199 @@
+        ...
+      </Button>
+    )
+  }
+}
+
+const usePaginationPages = (
+  currentPage: number,
+  totalPages: number | undefined,
+  maxButtons: number | undefined | false,
+  gotoPage: (page: number) => void,
+  createPageButton: (i: number) => ReactElement
+) => {
+  const [startEllipsisInput, setStartEllipsisInput] = useState(false)
+  const [endEllipsisInput, setEndEllipsisInput] = useState(false)
 
-  const pages: ReactElement[] = useMemo(() => {
-    const pages: ReactElement[] = []
+  return useMemo(() => {
+    if (totalPages === undefined) return []
+
+    let pages: ReactElement[] = []
+
+    // Create an array of all page buttons
     for (let i = 0; i < totalPages; i++) {
-      pages.push(
-        <Button key={i} onClick={() => setPage(i)} isActive={page === i}>
-          {i + 1}
-        </Button>
-      )
+      pages.push(createPageButton(i))
     }
-    return pages
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [page, totalPages])
 
-  return <ButtonGroup isAttached>{pages.map((page) => page)}</ButtonGroup>
+    if (!maxButtons || totalPages <= maxButtons) {
+      return pages
+    }
+
+    const startEllipsis = createEllipsisButton(
+      'start-ellipsis',
+      (e) => {
+        e.preventDefault()
+        setStartEllipsisInput(true)
+      },
+      startEllipsisInput,
+      (e) => {
+        if (e.key === 'Enter') {
+          const pageNumber = Number((e.target as HTMLInputElement).value)
+          gotoPage(pageNumber - 1)
+          setStartEllipsisInput(false)
+        }
+      },
+      () => setStartEllipsisInput(false)
+    )
+
+    const endEllipsis = createEllipsisButton(
+      'end-ellipsis',
+      (e) => {
+        e.preventDefault()
+        setEndEllipsisInput(true)
+      },
+      endEllipsisInput,
+      (e) => {
+        if (e.key === 'Enter') {
+          const pageNumber = Number((e.target as HTMLInputElement).value)
+          gotoPage(pageNumber - 1)
+          setEndEllipsisInput(false)
+        }
+      },
+      () => setEndEllipsisInput(false)
+    )
+
+    // Add ellipsis and slice the array accordingly
+    const sideButtons = 2 // First and last page
+    const availableButtons = maxButtons - sideButtons // Buttons we can distribute around the current page
+
+    if (currentPage <= availableButtons / 2) {
+      // Near the start
+      return [...pages.slice(0, availableButtons), endEllipsis, pages[totalPages - 1]]
+    } else if (currentPage >= totalPages - 1 - availableButtons / 2) {
+      // Near the end
+      return [pages[0], startEllipsis, ...pages.slice(totalPages - availableButtons, totalPages)]
+    } else {
+      // In the middle
+      const startPage = currentPage - Math.floor((availableButtons - 1) / 2)
+      const endPage = currentPage + Math.floor(availableButtons / 2)
+      return [pages[0], startEllipsis, ...pages.slice(startPage + 1, endPage), endEllipsis, pages[totalPages - 1]]
+    }
```

Comment:
There are a strange behavior if the max buttons is set to `1`... Shall we handle it? For example, not showing the Ellipsis buttons or  either show the page button with the previous/next buttons? 

![image](https://github.com/vocdoni/explorer/assets/16777076/4dcca5b4-dcf9-4458-b91a-a806944f937e)

Reply:
Not sure we should handle the "1" max buttons. Why do you want a single button for pagination?